[IMP] create JavaScript howtos (#3969)
odoo/documentation: content/developer/howtos/javascript_fields.rst

> ======

Reviewer: split this into 2 how to: one titled "Create a new field component" and another one titled "Subclass an existing field component"

Reviewer 2: An alternative could be to keep both sections on the same page but relabel the document to something like "Customize the display of a field", which provides two ways of doing so depending on your use case. Same thing for the views.

> Fields

Reviewer 2: As per the card's title (and because it's clearer).
```suggestion
Create a new field component
```

======

> This is probably a really common usecase: we want to display some information in a form view in a

Reviewer 2: ```suggestion
A common use case is to display some information in a form view in a
```
